pkg/ocmprovider/ocmprovider.go

```diff
@@ -12,6 +12,7 @@ import (
 
 	upgradev1alpha1 "github.com/openshift/managed-upgrade-operator/api/v1alpha1"
 	"github.com/openshift/managed-upgrade-operator/pkg/ocm"
+	"github.com/openshift/managed-upgrade-operator/pkg/ocmagent"
```

Comment:
Can we build this as a separated provider, called `ocmagentprovider`? 
So that we don't need to change anything within the existing `ocmprovider` and we can switch between them easily.

Reply:
And them, we can add another condition here 
https://github.com/openshift/managed-upgrade-operator/blob/master/pkg/specprovider/specprovider.go#L44

Reply:
but TBH, I am not sure if it is better or not, if the code are almost the same, there might be just a copy/paste of code blocks.

Reply:
I don't mind either way - whichever one you think is better, I am happy to do.